File: app/src/test/java/aoc2023/Day02Test.java
package aoc2023;

import static org.junit.jupiter.api.Assertions.assertEquals;

import aoc2023.day02.Day02;
import java.util.List;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestMethodOrder;
import utils.InputUtils;

@TestMethodOrder(MethodOrderer.MethodName.class)
public class Day02Test {

  private List<String> input = InputUtils.readInput("aoc2023/day02.txt");
  private List<String> inputExample = InputUtils.readInput("aoc2023/day02_example.txt");

  @Test
  public void testInputExamplePart1() {
    // when
    var result = Day02.part1(inputExample);

    // then
    assertEquals(8, result);
  }

  @Test
  public void testInputPart1() {
    // when
    var result = Day02.part1(input);

    // then
    assertEquals(2439, result);
  }

  @Test
  public void testInputExamplePart2() {
    // when
    var result = Day02.part2(inputExample);

    // then
    assertEquals(2286, result);
  }

  @Test
  public void testInputPart2() {
    // when
    var result = Day02.part2(input);

    // then
    assertEquals(63711, result);
  }
}

File: src/test/kotlin/aoc2022/Day02Test.kt
package aoc2022

import aoc2022.day02.Day02
import org.junit.jupiter.api.MethodOrderer
import org.junit.jupiter.api.TestMethodOrder
import utils.readInput
import kotlin.test.Test
import kotlin.test.assertEquals

@TestMethodOrder(
    MethodOrderer.Alphanumeric::class
)
internal class Day02Test {

    private val testInput = readInput("aoc2022/Day02_test.txt")
    private val input = readInput("aoc2022/Day02.txt")

    @Test
    internal fun testPart1() {
        // when
        val result = Day02.part1(testInput)

        // then
        assertEquals(15, result)

        // get solution
        println("Result of Day 02 - Part 1: ${Day02.part1(input)}")
    }

    @Test
    internal fun testPart2() {
        // when
        val result = Day02.part2(testInput)

        // then
        assertEquals(12, result)

        // get solution
        println("Result of Day 02 - Part 2: ${Day02.part2(input)}")
    }

}